analyses/tmb-compare-tcga/TMB_d3b_code/code/01_calculate_tmb_targetflexible_withbothexperstrt_and_cohort.py

```diff
@@ -0,0 +1,235 @@
+##  This script takes MAF file and  metadata  file with experimental strategy
+#   and target config file and computes TMB  scores for every samplename
+####### Author: Teja Koganti ##################
+# Inputs  - 1. MAF file - Consensus should be performed prior to using as INputs
+#               Must follow the MAF format. It's okay  if the first
+#               commented line is absent
+########### 2. metadata file - This should have  disease column and
+#               experimental strategy. Sample should not be  repeated under
+#               different experimental strategies
+########### 3. target config file - This should have two columns,
+#               one with experimental strategy in metadata  and another with the
+#               corresponding BED file for processing variant counts
+
+
+# python3 01_calculate_tmb_targetflexible.py \
+#   -i pbta-snv-consensus.maf
+#   -m pbta-histologies.tsv \
+#   -w ../inputs/target_cfg \
+#   -o ../output/pbta-snv-consensus-TMB_intarget.txt
+
+import subprocess
+import argparse
+import sys
+
+
+# This checks the packages to be installed if not already
+#   installed by user
+def install_package(package, package_list):
+    if package not in package_list:
+        print("Installing package " + package)
+        pip._internal.main(["install", package])
+
+
+###################################################################
+############# Checking if all packages are installed ##############
+
+reqs = subprocess.check_output([sys.executable, "-m", "pip", "freeze"])
+installed_packages = [r.decode().split("==")[0] for r in reqs.split()]
+
+needed_packages = [
+    "pandas",
+    "numpy",
+    "pybedtools",
+]
+
+for package in needed_packages:
+    install_package(package, installed_packages)
+
+##################################################################
```

Comment:
All code for this project should assume it is being run in the package docker container, which should already have these packages installed. If not, we will need to add them here https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/5b38033f25b64654b7a663b1b8c06398205a973c/Dockerfile#L312 (`pybedtools` is the only one not currently part of the container) I am happy to help if you need assistance with that section; we would want to specify version numbers for all packages and dependencies.

We do not want to encourage external installation, as that can lead to version conflicts.